pkg/agent/client_test.go

```diff
@@ -354,6 +357,18 @@ func newClosePacket(connID int64) *client.Packet {
 	}
 }
 
+func waitForConnectionDeletion(t *testing.T, testClient *Client, connID int64) {
+	t.Helper()
+	var ok bool
+	wait.PollImmediate(100*time.Millisecond, 3*time.Second, func() (bool, error) {
+		_, ok = testClient.connManager.Get(connID)
+		return !ok, nil
+	})
+	if ok {
```

Comment:
nit: wait.PollImmediat returns a timeout error if it times out, so I think this is usually done as
```suggestion
	if wait.PollImmediate(100*time.Millisecond, 3*time.Second, func() (bool, error) {
		_, ok = testClient.connManager.Get(connID)
		return !ok, nil
	}) != nil {
```

or if you prefer:

```suggestion
	err := wait.PollImmediate(100*time.Millisecond, 3*time.Second, func() (bool, error) {
		_, ok = testClient.connManager.Get(connID)
		return !ok, nil
	})
	if err != nil {
```

Reply:
Done